ui/hooks/metamask-notifications/useProfileSyncing.ts

```diff
@@ -0,0 +1,156 @@
+        await dispatch(performSignIn());
+      }
+
+      // set profile syncing to true
+      await dispatch(enableProfileSyncingAction());
+    } catch (e) {
+      // if an error occurs, we need to be sure that the
+      // profileSyncing is not enabled
+      await dispatch(disableProfileSyncingAction());
+
+      const errorMessage =
+        e instanceof Error ? e.message : JSON.stringify(e ?? '');
+      setError(errorMessage);
+    } finally {
+      setLoading(false);
+      dispatch(hideLoadingIndication());
+    }
+
+    setLoading(false);
+    dispatch(hideLoadingIndication());
+  }, [dispatch, isSignedIn]);
+
+  return { enableProfileSyncing, loading, error };
+}
+
+/**
+ * Custom hook to disable profile syncing. This hook handles the process of disabling notifications,
+ * disabling profile syncing, and signing out if MetaMetrics participation is not enabled.
+ *
+ * @returns An object containing the `disableProfileSyncing` function, current profile syncing state,
+ * loading state, and error state.
+ */
+export function useDisableProfileSyncing(): {
+  disableProfileSyncing: () => Promise<void>;
+  loading: boolean;
+  error: string | null;
+} {
+  const dispatch = useDispatch();
+  const isSignedIn = useSelector(selectIsSignedIn);
+  const isParticipateInMetaMetrics = useSelector(
+    selectParticipateInMetaMetrics,
+  );
+  const isMetamaskNotificationsEnabled = useSelector(
+    selectIsMetamaskNotificationsEnabled,
+  );
+
+  const { disableNotifications } = useDisableNotifications();
+
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const disableProfileSyncing = useCallback(async () => {
+    setLoading(true);
+    dispatch(showLoadingIndication());
+    setError(null);
+
+    try {
+      // if the notifications are enabled, disable them
+      // and set the states to false
+      if (isMetamaskNotificationsEnabled) {
+        await disableNotifications();
+      }
+
+      // disable profile syncing
+      await dispatch(disableProfileSyncingAction());
+
+      // if metametrics is not enabled, perform sign out
+      if (!isParticipateInMetaMetrics) {
+        // perform sign out
+        await dispatch(performSignOut());
+      }
+    } catch (e) {
+      const errorMessage =
+        e instanceof Error ? e.message : JSON.stringify(e ?? '');
+      setError(errorMessage);
+    } finally {
+      setLoading(false);
+      dispatch(hideLoadingIndication());
+    }
+
+    setLoading(false);
+    dispatch(hideLoadingIndication());
+  }, [dispatch, isSignedIn, isMetamaskNotificationsEnabled]);
+
+  return { disableProfileSyncing, loading, error };
+}
+
+/**
+ * Custom hook to check if profile syncing is enabled. This hook uses a selector to retrieve
+ * the profile syncing state from the Redux store.
+ *
+ * @returns An object containing the current state of profile syncing.
+ */
+export function useIsProfileSyncingEnabled(): {
```

Comment:
I see this hook defined here, but I also see this line of code below:
```
const isProfileSyncingEnabled = useSelector(selectIsProfileSyncingEnabled);
```

We could address that specific case, but it raises a question for me. I'm noticing that the use of hooks to wrap a single selector is new in our codebase. On first glance, it adds unneccessary code (e.g. why make a `useIsProfileSyncingEnabled` hooks when we can just directly use `selectIsProfileSyncingEnabled`). Could you share your thoughts on the benefits on the approach taken in this PR with these types of hooks?

Reply:
I agree. In fact, for now, it's nonsensical and a selector would be more than sufficient.